src/symbolize/gimli/libs_windows.rs

```diff
@@ -43,13 +42,79 @@ unsafe fn add_loaded_images(ret: &mut Vec<Library>) {
     }
 }
 
+// Safety: long_path should be null-terminated
+#[cfg(target_os = "cygwin")]
+unsafe fn get_posix_path(long_path: &[u16]) -> Option<OsString> {
+    use super::mystd::os::unix::ffi::OsStringExt;
+
+    unsafe extern "C" {
+        // Doc: https://cygwin.com/cygwin-api/func-cygwin-conv-path.html
+        // Src: https://github.com/cygwin/cygwin/blob/718a15ba50e0d01c79800bd658c2477f9a603540/winsup/cygwin/path.cc#L3902
+        // Safety:
+        // * `what` should be `CCP_WIN_W_TO_POSIX` here
+        // * `from` is `*const u16`, null-terminated, UTF-16 path
+        // * `to` is `*mut u8` buffer, the buffer size is `size`.
+        fn cygwin_conv_path(
+            what: libc::c_uint,
+            from: *const libc::c_void,
+            to: *mut libc::c_void,
```

Comment:
So now, our only question is left as "why not just declare `from` as `*const u16` and `to` as `*mut u8`*?" Those types are ABI compatible and it would prevent errors from casts.

Aside from that, this looks fine.